### PR TITLE
[Cleanup] Remove all traces of hbs

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Background and motivation:

ember-template-imports was originally intended to be an experiment for _all_ (and more!) possibilities outlined in https://github.com/emberjs/rfcs/pull/779 (and explored through this blog series: https://v5.chriskrycho.com/journal/ember-template-imports/ ). 

Of those possibilities, ember-template-imports implemented 2:
 - <code>hbs\`\`</code> - https://v5.chriskrycho.com/journal/ember-template-imports/part-3/ 
 - `<template>` - what was eventually chosen via [RFC  779](https://github.com/emberjs/rfcs/pull/779) 
 
 So, since RFC 779 chose `<template>`, we no longer need <code>hbs\`\`</code>, which this PR removes.

------------------

To help simplify the work of https://github.com/ember-template-imports/ember-template-imports/issues/143, we want to remove as much of the old stuff as possible before we extract the preprocessor and the babel plugin to their own locations.

This is most def a breaking change.
Though, there is no rush on cutting a release, and we could even wait until the extraction outlined in #143 is complete.

------------------

This PR is _just_ removal, to minimize the diff.
There are some obvious refactors / de-type unioning that can happen afterwards.

### CODE REVIEW

File: src/babel-plugin.js [hunk 1]
```diff
@@ -11,14 +10,6 @@ const { transformTemplateTag } = require('./template-tag-transform');
  * Its goal is to convert code like this:
  *
  * ```js
- * import { hbs } from 'ember-template-imports';
- *
- * const A = hbs(`A`, {...});
- * const B = [__GLIMMER_TEMPLATE(`B`, {...})];
```
Comment (dfreeman) [inline src/babel-plugin.js:17]: This line is still relevant, no?

File: src/parse-templates.ts [hunk 2]
```diff
@@ -286,89 +203,13 @@ export function parseTemplates(
...
-   */
-  function parseDynamicSegment(
-    results: TemplateMatch[],
-    template: string,
-    _startToken: RegExpMatchArray,
-    tokens: RegExpMatchArray[]
-  ) {
-    let stack = 1;
-
+  function parseTemplateLiteral(template: string, tokens: RegExpMatchArray[]) {
```
Comment (dfreeman) [inline src/parse-templates.ts:206]: I think this code predates anyone who still looks at this project, so I could just be misunderstanding, but why do we still need this function at all? Isn't "template literal" the term the implementation here uses to refer to the tagged-string version of templates that you're removing with this PR?

File: src/preprocessor-plugin.js [hunk 3]
```diff
@@ -61,37 +49,21 @@ module.exports = class TemplateImportPreprocessor {
...
-          Object.assign(config, this.templateLiteralConfig);
-        }
+    let compiled = stew.map(tree, `**/*.{gjs,gts}`, (string, relativePath) => {
+      let config = {
+        relativePath,
+        getTemplateLocalsRequirePath: this.getTemplateCompilerPath(),
+      };
 
-        return preprocessEmbeddedTemplates(string, config).output;
+      if (relativePath.match(/\.(gjs|gts)$/)) {
```
Comment (dfreeman) [inline src/preprocessor-plugin.js:61]: When will this condition ever not match?

File: src/parse-templates.ts (hunk 2 above)
Comment (NullVoxPopuli) [reply src/parse-templates.ts:206]: Template tags may exist with actual strings, such as within a REPL that demonstrates how to build a REPl with itself. anything in a string, we want to ignore.

File: src/babel-plugin.js (hunk 1 above)
Comment (NullVoxPopuli) [reply src/babel-plugin.js:17]: Yeah, but more importantly, the input in the comment doesn't match the output in the comment.

Will fix

File: src/preprocessor-plugin.js (hunk 3 above)
Comment (dfreeman) [reply src/preprocessor-plugin.js:61]: This conditional is directly tied to the line you changed above, turning `{js,gjs,ts,gts}` into just `{gjs,gts}`, so this isn't so much "refactoring" as "keeping the code consistent with the other changes you made". If you're planning on doing another pass in a separate PR, though, then I'm not going to lose sleep over it.

File: src/parse-templates.ts (hunk 2 above)
Comment (dfreeman) [reply src/parse-templates.ts:206]: So this is now doing the opposite of its name and _avoiding_ accidentally parsing string content as a template?

Comment (NullVoxPopuli) [reply src/parse-templates.ts:206]: exactly, yeah -- the function was pulling double duty previously

**Pull request merged**